frontend/src/pages/Browse/Browse.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import TreeMenu, {Item, TreeNodeInArray} from 'react-simple-tree-menu';
+import { Grid } from 'semantic-ui-react';
+import { SpecialistsList } from '../../components/SpecialistsList';
+
+export const Browse: FunctionComponent = () => {
+    const [menuNodes, setMenuNodes] = useState<MenuNode[]>();
+    const [selectedService, setSelectedService] = useState<number>();
+
+    useEffect(() => {
+        Promise.all([
+            fetch(process.env.REACT_APP_GET_All_TAGS_URL!),
+            fetch(process.env.REACT_APP_GET_All_SERVICES_URL!)
+        ])
+            .then(([getTagsResponse, getServicesResponse]: [Response, Response]) =>
+                Promise.all([getTagsResponse.json(), getServicesResponse.json()])
+            )
+            .then(([getTagsResponse, getServicesResponse]: [GetTagsResponse, GetServicesResponse]) => {
+                const tagsDto = getTagsResponse.tags;
+                const services = getServicesResponse.services.reduce<{ [tagId: number]: MenuNode[] }>(
+                    (accumulator, currentValue) => {
+                        accumulator[currentValue.tagId] = [
```

Comment:
I think this code can become a bit more readable if you use 'push' method here, something like:
accumulator[currentValue.tagId] = (accumulator[currentValue.tagId] || []).push(convertServiceToTreeNode(currentValue))